src/ripple/protocol/impl/Book.cpp

```diff
@@ -47,58 +47,30 @@ reversed(Book const& book)
     return Book(book.out, book.in);
 }
 
-/** Ordered comparison. */
-int
-compare(Book const& lhs, Book const& rhs)
-{
-    int const diff(compare(lhs.in, rhs.in));
-    if (diff != 0)
-        return diff;
-    return compare(lhs.out, rhs.out);
-}
-
 /** Equality comparison. */
 /** @{ */
 bool
 operator==(Book const& lhs, Book const& rhs)
 {
     return (lhs.in == rhs.in) && (lhs.out == rhs.out);
 }
-
-bool
-operator!=(Book const& lhs, Book const& rhs)
-{
-    return (lhs.in != rhs.in) || (lhs.out != rhs.out);
-}
 /** @} */
 
 /** Strict weak ordering. */
 /** @{ */
-bool
-operator<(Book const& lhs, Book const& rhs)
+std::weak_ordering
+operator<=>(Book const& lhs, Book const& rhs)
 {
-    int const diff(compare(lhs.in, rhs.in));
-    if (diff != 0)
-        return diff < 0;
-    return lhs.out < rhs.out;
-}
-
-bool
-operator>(Book const& lhs, Book const& rhs)
-{
-    return rhs < lhs;
-}
-
-bool
-operator>=(Book const& lhs, Book const& rhs)
-{
-    return !(lhs < rhs);
-}
-
-bool
-operator<=(Book const& lhs, Book const& rhs)
-{
-    return !(rhs < lhs);
+    auto const cmpIn{lhs.in <=> rhs.in};
+    if (cmpIn != std::weak_ordering::equivalent)
+    {
+        return cmpIn;
+    }
+    else
+    {
+        return lhs.out <=> rhs.out;
+    }
```

Comment:
It's up to you, but I think this is clearer:

```
    if (auto const c{lhs.in <=> rhs.in}; c != std::weak_ordering::equivalent)
        return c;

        return lhs.out <=> rhs.out;
```

Reply:
Sure. Fixed.